test/unittests/utils/test_filemanip.py

```diff
@@ -5,6 +5,140 @@
+    "dictionary, expected",
+    [
+        (
+            {
+                "TotalReadoutTime": 1,
+                "EstimatedTotalReadoutTime": "",
+                "PhaseEncodingSteps": "",
+                "PixelBandwidth": "",
+                "PhaseEncodingDirection": "j+",
+                "PhaseEncodingAxis": "",
+            },
+            [1, "j+"],
+        ),
+        (
+            {
+                "EstimatedTotalReadoutTime": 1,
+                "PhaseEncodingAxis": "j",
+            },
+            [1, "j+"],
+        ),
+        (
+            {
+                "PhaseEncodingSteps": 1,
+                "PixelBandwidth": 0.5,
+                "PhaseEncodingDirection": "j-",
+            },
+            [2, "j-"],
+        ),
+    ],
+)
+def test_extract_metadata_from_json_dwi(tmp_path, dictionary, expected):
+    """This function tests that the outputs of `extract_metadata_from_json` are what you'd expect in the case of DWI."""
+    import json
+
+    from clinica.utils.filemanip import (
+        extract_metadata_from_json,
+        handle_missing_keys_dwi,
+    )
+
+    with open(tmp_path / "metadata.json", "w") as outfile:
+        json.dump(dictionary, outfile)
+    assert (
+        extract_metadata_from_json(
+            tmp_path / "metadata.json",
+            [
+                "TotalReadoutTime",
+                "PhaseEncodingDirection",
+            ],
+            handle_missing_keys_dwi,
+        )
+        == expected
+    )
+
+
+@pytest.mark.parametrize(
+    "input_list, dictionary, error_type, error_log",
+    [
+        (
+            [
+                "TotalReadoutTime",
+                "PhaseEncodingDirection",
+            ],
+            {
+                "PhaseEncodingSteps": 1,
+                "PixelBandwidth": 0.5,
+            },
+            ClinicaException,
+            "Could not recover the PhaseEncodingDirection from JSON file.",
+        ),
+        (
+            [
+                "TotalReadoutTime",
+                "PhaseEncodingDirection",
+            ],
+            {
+                "PhaseEncodingDirection": "j+",
+            },
+            ClinicaException,
+            "Could not recover the TotalReadoutTime from JSON file.",
+        ),
+        (
+            [
+                "TotalReadoutTime",
+                "PhaseEncodingDirection",
+            ],
+            {
+                "PhaseEncodingSteps": 1,
+                "PixelBandwidth": 0,
+                "PhaseEncodingDirection": "j+",
+            },
+            ValueError,
+            "Pixel Bandwidth value is not valid.",
+        ),
+        (
+            [
+                "blabla",
+                "PhaseEncodingDirection",
+            ],
+            {
+                "PhaseEncodingDirection": "j+",
+            },
+            ValueError,
+            "Could not recover the missing keys {'blabla'} from JSON file.",
+        ),
+    ],
+)
+def test_crash_extract_metadata_from_json_dwi(
```

Comment:
I think we follow the more or less implicit convention of adding `errors` as a suffix:

```suggestion
def test_extract_metadata_from_json_dwi_errors(
```